packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ByteArray, Endian } from "../util/ByteArray.js";
+import { DataReader } from "../util/DataReader.js";
+import { DataWriter } from "../util/DataWriter.js";
+
+export interface HandshakeRequestPayload {
+    versions: ByteArray,
+    attMtu: number,
+    clientWindowSize: number,
+}
+
+export interface Header {
+    handshakeBit: number,
+    managementBit: number,
+    ackMsgBit: number,
+    endSegmentBit: number,
+    beginSegmentBit: number
+}
+
+export interface HandshakeRequest {
+    payload: HandshakeRequestPayload
+}
+
+export interface HandshakeResponsePayload {
+    finalVersion: number,
+    attMtu: number,
+    windowSize: number
+}
+
+export interface HandshakeResponse {
+    header: ByteArray,
+    payload: HandshakeResponsePayload
+}
+
+export interface PacketPayload {
+    ackNumber?: number,
+    sequenceNumber: number,
+    msgLength?: number,
+    segmentPayload?: ByteArray
+}
+
+export interface Packet {
+    header: Header,
+    payload: PacketPayload
+}
+
+export const enum HeaderBits {
+    HandshakeBit = 0b00000010,
+    ManagementMsg = 0b00000100,
+    AckMsg = 0b00010000,
+    EndSegment = 0b00100000,
+    BeginSegment = 0b10000000
+}
+
+export const enum Opcode {
+    HandshakeManagementOpcode = 0x6c,
+}
+
+const HANDSHAKE_HEADER = 0b01100101;
+
+export class BtpCodec {
+
+    static decodeHandshakeRequest(data: ByteArray): HandshakeRequest {
+        const reader = new DataReader(data, Endian.Little);
+
+        return {
+            payload: this.decodeRequestPayload(reader)
```

Comment:
I think then we can return payload directly. No need for an extra sub structure, or?

Reply:
Done